<!-- Copyright (C) 2019 Earth Rover Limited. All rights reserved. -->

<!-- EarthRover Scouting System: rosbag play for debug
  This file plays the rosbag files with certain parameters defined when calling it
  Author: David Llevat Pamies (dllevat@edgebrain.io)
  Date: 05/2019
-->

<launch>

  <!-- Parameters definition -->
  <arg name="bag_ipath"          default="/home/earth/earth_rover_ws/media/bags/" />
  <arg name="bag_opath"          default="/home/earth/earth_rover_ws/media/bags/" />
  <arg name="bag_required"      default="true" />
  <arg name="bag_output"        default="screen" />

  <arg name="bag_options" default="--pause --skip-empty=5" />

  <arg name="bag_topics_base"  default="/piksi_receiver/navsatfix_best_fix /piksi_attitude/baseline_heading /mti/sensor/imu /mti/sensor/magnetic /piksi_receiver/navsatfix_rtk_fix /piksi_receiver/navsatfix_rtk_float /piksi_receiver/navsatfix_spp" />

  <!-- Parameters setting -->
  <!-- Use sim time when playing back bag files with the clock flag -->
  <param name="use_sim_time" value="true" />

  <node name="player" type="play" pkg="rosbag" args="$(arg bag_options) --clock $(arg bag_ipath) --topics $(arg bag_topics_base)" required = "$(arg bag_required)" output = "$(arg bag_output)"/>

  <!-- recorder -->
  <node name="recorder" type="record" pkg="rosbag" args="--output-name $(arg bag_opath).bag --all" output = "screen"/>

</launch>
